aws-synthetics-canary/src/main/java/com/amazon/synthetics/canary/UpdateHandler.java

```diff
@@ -179,18 +178,23 @@
                     timeoutInSeconds = model.getRunConfig().getTimeoutInSeconds();
                 }
 
-                if (model.getRunConfig() != null && model.getRunConfig().getMemoryInMB() != null && memoryInMB != model.getRunConfig().getMemoryInMB()) {
+                if (model.getRunConfig().getMemoryInMB() != null && memoryInMB != model.getRunConfig().getMemoryInMB()) {
```

Comment:
The run config is still optional, right? So I think it makes sense to keep the `null` check.